rust/docs/async/Trait_EntityTypeAPI.adoc

```diff
@@ -0,0 +1,245 @@
+where
+     Self: Sync + 'async_trait,
+     'life0: 'async_trait,
+     'life1: 'async_trait,
+     'life2: 'async_trait,
+----
+
+.Code examples
+[source,rust]
+----
+entity_type.create(transaction).await;
+----
+
+[#_trait_EntityTypeAPI_method_get_supertype]
+==== get_supertype
+
+[source,rust]
+----
+fn get_supertype<'life0, 'life1, 'life2, 'async_trait>(
+    &'life0 self,
+    transaction: &'life1 Transaction<'life2>
+) -> Pin<Box<dyn Future<Output = Result<Option<EntityType>>> + Send + 'async_trait>>
+where
+     Self: Sync + 'async_trait,
+     'life0: 'async_trait,
+     'life1: 'async_trait,
+     'life2: 'async_trait,
+----
+
+Retrieves the most immediate supertype of the ``EntityType``.
+
+[caption=""]
+.Input parameters
+[cols="~,~,~"]
+[options="header"]
+|===
+|Name |Description |Type
+a| `transaction` a| The current transaction a| `&'life1 Transaction<'life2>` 
+|===
+
+.Returns
+[source,rust]
+----
+Pin<Box<dyn Future<Output = Result<Option<EntityType>>> + Send + 'async_trait>>
+where
+     Self: Sync + 'async_trait,
+     'life0: 'async_trait,
+     'life1: 'async_trait,
+     'life2: 'async_trait,
+----
+
+.Code examples
+[source,rust]
+----
+entity_type.get_supertype(transaction).await;
+----
+
+[#_trait_EntityTypeAPI_method_set_supertype]
+==== set_supertype
+
+[source,rust]
+----
+fn set_supertype<'life0, 'life1, 'life2, 'async_trait>(
+    &'life0 mut self,
+    transaction: &'life1 Transaction<'life2>,
+    supertype: EntityType
+) -> Pin<Box<dyn Future<Output = Result> + Send + 'async_trait>>
+where
+     Self: Send + 'async_trait,
+     'life0: 'async_trait,
+     'life1: 'async_trait,
+     'life2: 'async_trait,
+----
+
+Sets the supplied ``EntityType`` as the supertype of the current ``EntityType``.
+
+[caption=""]
+.Input parameters
+[cols="~,~,~"]
+[options="header"]
+|===
+|Name |Description |Type
+a| `transaction` a| The current transaction a| `&'life1 Transaction<'life2>` 
+a| `supertype` a| The ``EntityType`` to set as the supertype of this ``EntityType`` a| `EntityType` 
+|===
+
+.Returns
+[source,rust]
+----
+Pin<Box<dyn Future<Output = Result> + Send + 'async_trait>>
+where
+     Self: Send + 'async_trait,
+     'life0: 'async_trait,
+     'life1: 'async_trait,
+     'life2: 'async_trait,
+----
+
+.Code examples
+[source,rust]
+----
+entity_type.set_supertype(transaction, super_entity_type).await;
+----
+
+[#_trait_EntityTypeAPI_method_get_supertypes]
+==== get_supertypes
+
+[source,rust]
+----
+fn get_supertypes(
+    &self,
+    transaction: &Transaction<'_>
+) -> Result<BoxStream<'_, Result<EntityType>>>
+----
+
+Retrieves all supertypes of the ``EntityType``.
+
+[caption=""]
+.Input parameters
+[cols="~,~,~"]
+[options="header"]
+|===
+|Name |Description |Type
+a| `transaction` a| The current transaction a| `&Transaction<'_>` 
+|===
+
+.Returns
+[source,rust]
+----
+Result<BoxStream<'_, Result<EntityType>>>
+----
+
+.Code examples
+[source,rust]
+----
+entity_type.get_supertypes(transaction).await;
```

Comment:
Perhaps out of scope, but the examples should show retrieving the items out of the stream.

Reply:
Yeah, we should do it later.